surfinBH/_fit_evaluators/fit_7dq4Emri.py

```diff
@@ -0,0 +1,267 @@
+        """ Loads fits from h5file and returns a dictionary of fits. """
+        fits = {}
+        for key in ['mf']:
+            fits[key] = self._load_scalar_fit(fit_key=key, h5file=h5file)
+        for key in ['chif']:
+            fits[key] = self._load_vector_fit(key, h5file)
+        return fits
+
+    #-------------------------------------------------------------------------
+    def _get_fit_params(self, x, fit_key):
+        """ Transforms the input parameter to fit parameters for the 7dq4Emri 
+    model.
+    That is, maps from
+    x = [q, chiAx, chiAy, chiAz, chiBx, chiBy, chiBz]
+    fit_params = [np.log(q), chiAx, chiAy, chiHat, chiBx, chiBy, chi_a]
+
+    chiHat is defined in Eq.(3) of 1508.07253, but with chiAz and chiBz instead
+    of chiA and chiB.
+    chi_a = (chiAz - chiBz)/2.
+        """
+        q, chiAz, chiBz = x[0], x[3], x[6]
+        eta = q/(1.+q)**2
+        chi_wtAvg = (q*chiAz+chiBz)/(1.+q)
+        chiHat = (chi_wtAvg - 38.*eta/113.*(chiAz + chiBz))/(1. - 76.*eta/113.)
+        chi_a = (chiAz - chiBz)/2.
+
+        fit_params = [np.log(q), x[1], x[2], chiHat, x[4], x[5], chi_a]
+        return fit_params
+
+    #-------------------------------------------------------------------------
+    def _generate_random_params_for_tests(self):
+        """ Generate random parameters to use in tests.
+        """
+        # Generate params randomly within allowed values. NRSur7dq4EmriRemnant 
+        # is valid at arbitrary mass ratios, but here an upper value is needed.
+        q = 10**np.random.uniform(0, 10)
+        chiAmag= np.random.uniform(0, self.hard_param_lims['chiAmag'])
+        chiBmag= np.random.uniform(0, self.hard_param_lims['chiBmag'])
+        if self.aligned_spin_only:
+            chiAph, chiBph = 0, 0
+            chiAth, chiBth = np.random.choice([0, np.pi]), \
+                np.random.choice([0, np.pi])
+        else:
+            chiAth = np.arccos(np.random.uniform(-1., 1.))
+            chiBth = np.arccos(np.random.uniform(-1., 1.))
+            chiAph = np.random.uniform(0, 2*np.pi)
+            chiBph = np.random.uniform(0, 2*np.pi)
+
+        chiA = [chiAmag*np.sin(chiAth)*np.cos(chiAph),
+                chiAmag*np.sin(chiAth)*np.sin(chiAph),
+                chiAmag*np.cos(chiAth)]
+
+        chiB = [chiBmag*np.sin(chiBth)*np.cos(chiBph),
+                chiBmag*np.sin(chiBth)*np.sin(chiBph),
+                chiBmag*np.cos(chiBth)]
+
+        return q, chiA, chiB
+
+    #-------------------------------------------------------------------------
+    def _eval_wrapper(self, fit_key, q, chiA, chiB, **kwargs):
+        """ Evaluates the NRSur7dq4EmriRemnant model. """
+        chiA = np.array(chiA)
+        chiB = np.array(chiB)
+
+        # Warn/Exit if extrapolating
+        allow_extrap = kwargs.pop('allow_extrap', False)
+        if allow_extrap: warnings.warn("""Optional argument 'allow_extrap' is 
+        unused for this model. It works at arbitrarily large mass ratios.""")
```

Comment:
Always go to a new line and do an indent after an if condition. Very hard to read otherwise.
"arbitrarily large mass ratios" --> arbitrary mass ratios and spins

Reply:
Done